File: pcaixa/tests/MoedaTest.php
<?php
use PHPUnit\Framework\TestCase;

use Caixa\Moeda;

class MoedaTest extends TestCase {
    
    /**
     * @covers Moeda
     */
	public function testConverterReaisEmDolar()
	{
		$moeda = new Moeda(1,'BRL');
		$this->assertEquals(1/5,$moeda->converterPara('USD'));
	}
	
	/**
	 * @covers Moeda
	 */
	public function testConverterReaisEmReais()
	{
	    $moeda = new Moeda(1100,'BRL');
	    $this->assertEquals(1100,$moeda->converterPara('BRL'));
	}
	
	/**
	 * @covers Moeda
	 */
	public function testFormatarReais()
	{
	    $moeda = new Moeda(1,'BRL');
	    $this->assertEquals('R$ 1,00', $moeda->exibir());
	}
	
	/**
	 * @covers Moeda
	 */
	public function testSomarMoedas()
	{
	    $real = new Moeda(10,'BRL');
	    $dolar = new Moeda(20, 'USD');
	    $this->assertEquals(110, $real->somar($dolar));
	}
}

File: jcaixa/src/jcaixa/MoedaTest.java
package jcaixa;

import junit.framework.TestCase;

public class MoedaTest extends TestCase {
	public void testConverterReaisEmDolar()
	{
		Moeda moeda = new Moeda(1,"BRL");
		assertEquals((float)1/5,moeda.converterPara("USD"));
	}
	
	public void testFormatarReais()
	{
	    Moeda moeda = new Moeda(1,"BRL");
	    assertEquals("R$ 1,00", moeda.exibir());
	}
	
	public void testSomarMoedas()
	{
	    Moeda real = new Moeda(10, "BRL");
	    Moeda dolar = new Moeda(20, "USD");
	    assertEquals(110, (int) real.somar(dolar));
	}
	
}
